### PR TITLE
Boot 3 Upgrade

### PR DESCRIPTION
**Pull request by BoykoAlex**

### CODE REVIEW

File: src/main/resources/META-INF/rewrite/spring-data3-upgrade.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#
+---
+########################################################################################################################
+# Spring Data 3.0
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.spring.boot3.data.UpgradeSpringData_3_0
+displayName: Upgrade to Spring Data 3.0
+description: 'Upgrade to Spring Data to 3.0 from any prior version.'
+recipeList:
+  - org.openrewrite.java.spring.boot3.data.SwitchToJakarta
```
Comment (pway99) [inline src/main/resources/META-INF/rewrite/spring-data3-upgrade.yml:24]: rewrite-migrate-java has a JavaxToJakarta recipe which can be used instead
https://github.com/openrewrite/rewrite-migrate-java/blob/main/src/main/resources/META-INF/rewrite/javax-to-jakarta.yml

Created an issue for adding a new recipe to support the xml-binding 4.0.0 api.
https://github.com/openrewrite/rewrite-migrate-java/issues/110

**Comment by BoykoAlex** [comment]
should include https://github.com/spring-projects-experimental/spring-boot-migrator/issues/164

**Comment by BoykoAlex** [comment]
@pway99 updated and rebased

**Comment by pway99** [comment]
Thanks @BoykoAlex.  I merged this in along with a TODO comment for rewrite-migrate-java Java17 f3e98e195bd5ecd7fd1fa8ea1251d4f0ba241a78

**Pull request closed without merging**